### ISSUE TITLE
Improper `too-many-function-args` for decorated function

### ISSUE THREAD

**Issue by angelolovatto** [PyCQA/pylint#4386]
### Steps to reproduce

The following are the contents of `variadic.py`.

```python
# pylint:disable=missing-docstring
from __future__ import annotations
import functools
from typing import Callable, Union


def variadic(func: Callable[[int], int]):
    @functools.wraps(func)
    def wrapped(*integers: int) -> Union[int, tuple[int, ...]]:
        result = tuple(func(i) for i in integers)
        return result[0] if len(result) == 1 else result

    return wrapped


@variadic
def double(integer: int) -> int:
    return integer * 2


def main():
    doubled = double(1, 2, 3)
    print(doubled)


if __name__ == "__main__":
    main()
```

### Current behavior

Result of `pylint variadic.py`:
```
************* Module variadic
variadic.py:22:14: E1121: Too many positional arguments for function call (too-many-function-args)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

Line 22 shouldn't raise `too-many-function-args`, as `double` has been decorated to handle a variable number of arguments.

This is especially troublesome as I have a similar setup that uses such a function across several modules.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.4
astroid 2.5.3
Python 3.8.2 (default, Jun  8 2020, 10:58:26) 
[Clang 11.0.3 (clang-1103.0.32.62)]
```

**Comment by hippo91** [comment]
@angelolovatto thanks for your report. While I agree that it is definitely a false positive bug I think that it will require a lot of work in astroid and pylint to be fixed.
